test_e2e_local.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source common.sh
+
+export TRACE=1
+export GO111MODULE=on
+
+fetch_tools
+install_kind
+build_kb
+
+setup_envs
+
+export KIND_CLUSTER=local-kubebuilder-e2e
+if ! kind get clusters | grep -q $KIND_CLUSTER ; then
+    source "$(pwd)/scripts/setup.sh" ${KIND_K8S_VERSION} $KIND_CLUSTER
+    docker pull gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0
+    kind load --name $KIND_CLUSTER docker-image gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0
+fi
+
+kind export kubeconfig --kubeconfig $tmp_root/kubeconfig --name $KIND_CLUSTER
+export KUBECONFIG=$tmp_root/kubeconfig
+
```

Comment:
We need the trap to ensure that all will be cleaned if an issue is faced.

```shell
# remove running containers on exit
function cleanup() {
    kind delete cluster
}

trap cleanup EXIT
``

Reply:
I found that on my machine, standing up the kind cluster a significant chunk of the total test run time. So for the local script I added in an override that leaves the kind cluster in place between runs: the tests seem very good at either cleaning up after themselves or not conflicting with each other, or both.